### ISSUE TITLE
Proposal: Change how API endpoint configuration works

### ISSUE THREAD

**Issue by chaudum** [crate/croud#268]
At the moment, there are "regions" and "environments" hard coded into the configuration. This means, that an API endpoint has to be configured using `croud config set --region REGION --env ENVIRONMENT`. The API endpoint is not transparent, though.

I propose a breaking change on how configuration works in order to properly solve the issue mentioned above.

### Default configuration
The configuration does not have "contexts" (environments) and "regions" any more, but is based on "profiles" which have an API "endpoint", a "format", and an "auth-token" assigned.
The top level "current-profile" defines which profile to use by default, if not explicitly specified using `--profile PROFILE`.
The default configuration contains 3 profiles with the permanent API endpoints.

```yaml
default-format: table
current-profile: bregenz.a1
profiles:
  bregenz.a1:
    endpoint: https://bregenz.a1.cratedb.cloud
    format: table
    auth-token: {}
  eastus2.azure:
    endpoint: https://eastus2.azure.cratedb.cloud
    format: table
    auth-token: {}
  westeurope.azure:
    endpoint: https://westeurope.azure.cratedb.cloud
    format: table
    auth-token: {}
```

### Configuration class

```py
class Configuration:

    def load(self) -> None:
        ...

    def dump(self) -> None:
        ...

    def add_profile(self, profile: str, **kwargs) -> None:
        ...

    def remove_profile(self, profile: str) -> None:
        ...

    def use_profile(self, profile: str) -> None:
        ...
```

### Commands

```console
$ croud config show
// prints full contents of configuration file (YAML)

$ croud config current-profile
// prints value of current-profile

$ croud config use-profile PROFILE
// sets value of current-profile

$ croud config add-profile PROFILE --endpoint ENDPOINT [--format FORMAT]
// adds a new profile object

$ croud config remove-profile PROFILE [--yes]
// removes an existing profile object
```

When invoking a specific command, the `current-profile` value can be overwritten using `--profile PROFILE`:

```console
$ croud clusters list --profile westeurope.azure --region eastus.azure
// use the API endpoint in westeurope to list resources deployed in eastus
```

**Comment by MarkusH** [comment]
Definitely :+1: from me. I think it would be worth noting that the profile name can be a (more or less) arbitrary string. The profile name doesn't need to correlate in any way to the endpoint. E.g. if my production cluster is in westeurope.azure and my staging cluster in eastus2.azure, I could name the profiles "production" and "staging" with the corresponding endpoints.

I think the command `croud config show` should hide or mask the value for the `auth-token`. If one wants to see its value they can look it up in the config file, but printing it out to stdout seems dangerous.

**Comment by chaudum** [comment]
> Definitely +1 from me. I think it would be worth noting that the profile name can be a (more or less) arbitrary string. The profile name doesn't need to correlate in any way to the endpoint. E.g. if my production cluster is in westeurope.azure and my staging cluster in eastus2.azure, I could name the profiles "production" and "staging" with the corresponding endpoints.

Definitely. We would need to document that properly. -- /cc @crate/docs 

> I think the command `croud config show` should hide or mask the value for the `auth-token`. If one wants to see its value they can look it up in the config file, but printing it out to stdout seems dangerous.

Good idea.

**Comment by chaudum** [comment]
Implemented with #269

**Comment by mechanomi** [comment]
cc @crate/tech-writing :)
